documentation/source/coroutines.rst

```diff
@@ -151,3 +151,35 @@ they'd naturally end.
         if not isclose(edge_time_ns, start_time_ns + 4.0):
             raise TestFailure("Expected a period of 4 ns")
 
+
+Async functions
+---------------
+
+Python 3.5 introduces ``async`` functions, which provide an alternative syntax.
+For example:
+
+.. code-block:: python
+
+    @cocotb.coroutine
+    async def wait_10ns():
+        cocotb.log.info("About to wait for 10ns")
+        await Timer(10000)
+        cocotb.log.info("Simulation time has advanced by 10 ns")
+
+To wait on a trigger or a nested coroutine, these use ``await`` instead of ``yield``.
```

Comment:
Could you add a sentence what happens if it is allowed to mix old and new syntax, or what happens if you do so? (i.e. `yield RisingEdge(dut.clk)` within a `async def` coroutine).

Reply:
Expanded the docs considerably